### DISCUSSION TITLE
Can I notify a class instance to make the view changed when any of it's fields has changed?

### DISCUSSION THREAD

**Discussion by keynes1981** [AvaloniaUI/Avalonia#15803] [Q&A] [answered]
I have a Test instance, when it's fields has changed in the command, Can I notify the instance to make the view changed?  I cannot use CommunityToolkit.Mvvm
public class MainWindowViewModel : ViewModelBase
{
    public Test Test { get; set; }
    public MainWindowViewModel()
    {
        Test = new Test();

        OKComand = ReactiveCommand.Create<string>((val) =>
        {
            Test.Id = 1111;
            Test.Title = "Hello World!";

            RaisePropteryChanged(nameof(Test));
        });
    }
    public ICommand? OKComand { get; set; }

}

public class Test
{
    public int Id { get; set; } = 123;
    public string Title { get; set; } = "Hello";
}

**Comment by timunie** [comment, marked as answer]
No, that's not working. Either you wrap your item in a helper class or replace the entire test instance afaik

**Comment by thevortexcloud** [comment]
I would argue you are not following the correct pattern here. Generally, only view models should be exposed to the view. If you need to expose a non view model, it should be wrapped by a view model. EG You might have a `Car` model class, and then a `CarViewModel` that accepts a `Car` in it's constructor and uses `Car` as a backing variable/to initialise its values. The other advantage of doing that is it lets you customise the presentation somewhat without messing up your data layer.
